sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -71,7 +72,15 @@ class QueryExecution(
 
   lazy val analyzed: LogicalPlan = executePhase(QueryPlanningTracker.ANALYSIS) {
     // We can't clone `logical` here, which will reset the `_analyzed` flag.
-    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
+    sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker) match {
+      case c: Command => c
```

Comment:
Why do we leave out the root node Command? Can we remove the command execution logic in `Dataset.logicalPlan` and execute all the commands here?

Reply:
I want to maintain the Command here and ensure its behavior.

Reply:
OK. Let's unify the behavior eagerly execute the commands.